### PR TITLE
unify sudo and do_execute methods

### PR DESCRIPTION
**Pull request by bparees**
previously the do_execute method was not handling the exception that was
thrown when an ssh command returned a non-zero rc.

### CODE REVIEW

**Comment by bparees** [comment]
@danmcp ptal.  obviously this should NOT go in until the release dust settles.

this will fix the problem where we currently exit out early and don't gather artifacts.

**Comment by bparees** [comment]
Note that i'm not sure why we had separate sudo and do_execute methods doing mostly, but not quite, the same thing previously, so there might be some history i'm missing/screwing up with this change.

**Comment by bparees** [comment]
@mfojtik this should fix artifact gathering on failure for PR tests.

File: lib/vagrant-openshift/helper/command_helper.rb [hunk 1]
```diff
@@ -21,13 +21,22 @@ module Openshift
     module CommandHelper
 
       def sudo(machine, command, options={})
+        # this flag is respected by the ssh machine.communicator
+        options[:sudo]=true
+        do_execute(machine,command,options)
```
Comment (danmcp) [inline lib/vagrant-openshift/helper/command_helper.rb]: Use machine, command, options spacing to be consistent

File: lib/vagrant-openshift/helper/command_helper.rb [hunk 2]
```diff
@@ -37,7 +46,7 @@ def sudo(machine, command, options={})
               machine.env.ui.info "Retrying. Attempt ##{retry_count} with timeout #{options[:timeout]}"
             end
             Timeout::timeout(options[:timeout]) do
-              rc = machine.communicate.sudo(command) do |type, data|
```
Comment (danmcp) [inline lib/vagrant-openshift/helper/command_helper.rb]: Where is options[:sudo] being used?

File: lib/vagrant-openshift/helper/command_helper.rb [hunk 3]
```diff
@@ -21,13 +21,22 @@ module Openshift
     module CommandHelper
 
       def sudo(machine, command, options={})
+        # this flag is respected by the ssh machine.communicator
+        options[:sudo]=true
```
Comment (danmcp) [inline lib/vagrant-openshift/helper/command_helper.rb]: Use
options[:sudo] = true
spacing to be consistent.

**Comment by bparees** [comment]
@danmcp reworked.

**Pull request merged**